src/ttsim/interface_dag_elements/fail_if.py

```diff
@@ -346,11 +346,13 @@ def environment_is_invalid(
     )
 
     flat_policy_environment = dt.flatten_to_tree_paths(policy_environment)
-    paths_with_incorrect_leaf_names = ""
-    for p, f in flat_policy_environment.items():
-        if hasattr(f, "leaf_name") and p[-1] != f.leaf_name:
-            paths_with_incorrect_leaf_names += f"    {p}\n"
-    if paths_with_incorrect_leaf_names:
+    incorrect_paths = [
+        f"    {p}\n"
+        for p, f in flat_policy_environment.items()
+        if hasattr(f, "leaf_name") and p[-1] != f.leaf_name
+    ]
+    if incorrect_paths:
+        paths_with_incorrect_leaf_names = "".join(incorrect_paths)
         msg = (
```

Comment:
```suggestion
    paths_with_incorrect_leaf_names = [
        f"    {p}"
        for p, f in flat_policy_environment.items()
        if hasattr(f, "leaf_name") and p[-1] != f.leaf_name
    ]
    if paths_with_incorrect_leaf_names:
        msg = (
```

And then use "\n".join(paths_with_incorrect_leaf_names) below, cannot comment there.